types/src/crypto/error.rs

```diff
@@ -36,3 +58,47 @@ impl From<SignatureError> for Error {
         Error::SignatureError(error)
     }
 }
+
+/// Cryptographic errors extended with some additional variants.
+#[cfg(any(feature = "crypto_std", test))]
+#[derive(Debug, Error)]
+pub enum ErrorExt {
```

Comment:
I wonder if `IoCryptoError` would be more descriptive: The additional variants all seem to be I/O-related?

Reply:
I think `FromPem` and `FromDer` can also happen because of parsing issues - and that might theoretically happen without I/O. But I'm open to naming it something more descriptive.

Reply:
Makes sense. I'm fine with `ErrorExt`, and it's not a blocker; but the name makes me think of an [extension trait](https://rust-lang.github.io/rfcs/0445-extension-trait-conventions.html) rather than an `enum`.

Reply:
It's kind of an extension enum :wink: But it's a good point that it's usually used in different contexts. I don't really have any ideas for a better name, though.